src/Downloader.ts

```diff
@@ -11,35 +11,44 @@ import axios, { AxiosRequestConfig } from 'axios';
 import imageminPngquant from 'imagemin-pngquant';
 import imageminGifsicle from 'imagemin-gifsicle';
 import imageminJpegoptim from 'imagemin-jpegoptim';
+import imageminWebp from 'imagemin-webp';
 
 import {
-  MIME_IMAGE_REGEX,
   normalizeMwResponse,
   objToQueryString,
   readFilePromise,
-  URL_IMAGE_REGEX,
   DB_ERROR,
   WEAK_ETAG_REGEX,
   writeFilePromise,
   renderArticle,
-  stripHttpFromUrl
+  stripHttpFromUrl,
+  isBitmapImageMimeType,
+  isImageUrl,
+  isWebpCandidateImage,
 } from './util';
 import S3 from './S3';
 import { Dump } from './Dump';
 import logger from './Logger';
 import MediaWiki from './MediaWiki';
 
 
-const imageminOptions = {
+const imageminOptions = new Map();
+imageminOptions.set('image/jpeg', {
+  plugins: [
+    imageminJpegoptim({ max: 60, stripAll: true }),
+  ]
+})
+imageminOptions.set('image/png', {
   plugins: [
-    // imageminOptiPng(),
     imageminPngquant({ speed: 3, strip: true, dithering: 0 }),
     imageminAdvPng({ optimizationLevel: 4, iterations: 5 }),
-    imageminJpegoptim({ max: 60, stripAll: true }),
-    // imageminJpegtran(),
+  ]
+})
+imageminOptions.set('image/gif', {
+  plugins: [
     imageminGifsicle({ optimizationLevel: 3, colors: 64 }),
-  ],
-};
+  ]
+})
```

Comment:
All of this is wrong, it was proper in last review. Please bring it back and just make small adaptation for imagemin-gif.